### PR TITLE
Extend Manifest installation consistency check with deployment availability check

### PR DESCRIPTION
**Pull request by khlifi411**
**Proposed changes**

- Implement deployment availability check

Fix: #415

### CODE REVIEW

File: internal/manifest/v1beta1/ready_check.go [hunk 1]
```diff
@@ -56,3 +71,36 @@ func (c *ManifestCustomResourceReadyCheck) Run(
 
 	return nil
 }
+
+func checkDeploymentState(resources []*resource.Info) (bool, error) {
+	var deploy *appsv1.Deployment
+	for _, res := range resources {
+		typedObject := kube.AsVersioned(res)
```
Comment (jakobmoellersap) [inline internal/manifest/v1beta1/ready_check.go:78]: this is a helm dependency we do not need I think. IF you bind the GVK anyway against appsv1, you can use the typedObject directly with the default converter from the scheme

File: Makefile [hunk 2]
```diff
@@ -179,3 +179,12 @@ lint: ## Run golangci-lint against code.
...
 	kubebuilder edit --plugins grafana.kubebuilder.io/v1-alpha
 	mv grafana/* config/grafana
+
+CRD_FILE_PATH ?= ./config/samples/tests/crds/operator.kyma-project.io_samplecrd.yaml
+ARCHIVES_DST ?= ./pkg/test_samples/oci
+HELM_CHART_PATH ?= ./pkg/test_samples/helm
+OCI_GEN_SCRIPT ?= $(ARCHIVES_DST)/generate_oci_archives.sh
+
+.PHONY: gen-oci-archives
+gen-oci-archives: ## Generate OCI Layers archives used for testing
```
[the remark this answers is not in the document]
Comment (khlifi411) [reply Makefile:189]: This is to make generating the archives used to test the OCI scenarios for the Manifest reconciler straight-forward. I will add some comments to the Make target to explain when to regenerate them 👍

File: internal/manifest/v1beta1/ready_check.go [hunk 3]
```diff
@@ -56,3 +68,28 @@ func (c *ManifestCustomResourceReadyCheck) Run(
...
+			break
+		}
+	}
+	if !found {
+		return false, ErrDeploymentResNotFound
+	}
+	if deploy.Spec.Replicas != nil && *deploy.Spec.Replicas != deploy.Status.ReadyReplicas {
+		return false, nil
+	}
+	availableCond := deploymentutil.GetDeploymentCondition(deploy.Status, appsv1.DeploymentAvailable)
```
[the remark this answers is not in the document]
Comment (khlifi411) [reply internal/manifest/v1beta1/ready_check.go:83]: Yes. That's a bug. Fixed it [here](https://github.com/kyma-project/lifecycle-manager/pull/419/commits/d6a5e4d56a185072d44dd039baf7959160550e46)

**Review by jakobmoellersap — APPROVED** [review]

**Pull request merged**